core/src/test/java/com/sap/ai/sdk/core/client/DeploymentUnitTest.java

```diff
@@ -195,4 +208,61 @@ void testDeleteAiDeployment() {
     // cloudSdkCustomFields
     assertThat(deployment.getCustomField("targetStatus")).isEqualTo("DELETED");
   }
+
+  @Test
+  void getDeploymentById() {
+    wireMockServer.stubFor(
+        get(urlPathEqualTo("/lm/deployments/db1d64d9f06be467"))
+            .withHeader("AI-Resource-Group", equalTo("default"))
+            .willReturn(
+                aResponse()
+                    .withStatus(HttpStatus.SC_OK)
+                    .withHeader("content-type", "application/json")
+                    .withBody(
+                        """
+                        {
+                           "configurationId": "dd80625e-ad86-426a-b1a7-1494c083428f",
+                           "configurationName": "orchestration",
+                           "createdAt": "2024-08-05T16:17:29Z",
+                           "deploymentUrl": "https://api.ai.intprod-eu12.eu-central-1.aws.ml.hana.ondemand.com/v2/inference/deployments/db1d64d9f06be467",
+                           "details": {
+                             "resources": {
+                               "backend_details": {}
+                             },
+                             "scaling": {
+                               "backend_details": {}
+                             }
+                           },
+                           "id": "db1d64d9f06be467",
+                           "lastOperation": "CREATE",
+                           "latestRunningConfigurationId": "dd80625e-ad86-426a-b1a7-1494c083428f",
+                           "modifiedAt": "2024-08-26T12:43:18Z",
+                           "scenarioId": "orchestration",
+                           "startTime": "2024-08-05T16:18:41Z",
+                           "status": "RUNNING",
+                           "submissionTime": "2024-08-05T16:17:40Z",
+                           "targetStatus": "RUNNING"
+                        }
+                        """)));
+    final AiDeploymentResponseWithDetails deployment =
+        new DeploymentApi(getClient(destination)).deploymentGet("default", "db1d64d9f06be467");
+
+    assertThat(deployment.getConfigurationId()).isEqualTo("dd80625e-ad86-426a-b1a7-1494c083428f");
+    assertThat(deployment.getConfigurationName()).isEqualTo("orchestration");
+    assertThat(deployment.getCreatedAt()).isEqualTo("2024-08-05T16:17:29Z");
+    assertThat(deployment.getDeploymentUrl())
+        .isEqualTo(
+            "https://api.ai.intprod-eu12.eu-central-1.aws.ml.hana.ondemand.com/v2/inference/deployments/db1d64d9f06be467");
+    assertThat(deployment.getId()).isEqualTo("db1d64d9f06be467");
```

Comment:
Missing deployment.getDetails()

Reply:
Deserialisation of details.backend_details needs fixing. But, I have added some assertions that works with the current behaviour.

[Existing ticket to fix key issue](https://jira.tools.sap/browse/AIWDF-2124)

Reply:
> Deserialisation of details.backend_details needs fixing

Can you elaborate on what exactly requires fixing with "deserialization"? There is no exception or anything, right? Or is there?

Reply:
To be exact, the incoming response contains key backend_details and spec defines backendDetails as the expected one. No exception is raised and then data is available as a custom field.

I have also included assertions for data returned from deployment.getDetails()